pkg/tsdb/sqleng/sql_engine.go

```diff
@@ -29,7 +31,7 @@ var XormDriverMu sync.RWMutex
 // MetaKeyExecutedQueryString is the key where the executed query should get stored
 const MetaKeyExecutedQueryString = "executedQueryString"
 
-var ErrConnectionFailed = errors.New("failed to connect to server - please inspect Grafana server log for details")
+var ErrConnectionFailed = errutil.NewBase(errutil.StatusInternal, "sql.connection-error")
```

Comment:
nit: Based on my understanding of the guidelines below, `"sql.connection-error"` should be `"sqleng.connectionError"`. I did notice that a lot of the error identifiers throughout the code base aren't camelCased though.

> The messageID is constructed as `<servicename>.<errorIdentifier>` where the `<servicename>` corresponds to the root service directory per [the package hierarchy](https://github.com/grafana/grafana/blob/main/contribute/backend/package-hierarchy.md) and `<errorIdentifier>` is a camelCased short identifier that identifies the specific category of errors within the service.

https://github.com/grafana/grafana/blob/main/contribute/backend/errors.md